code/game/objects/structures/sensor_tower.dm

```diff
@@ -107,42 +109,57 @@
 	if(user.do_actions)
 		user.balloon_alert(user, "You are already doing something!")
 		return
-	if(user.faction != FACTION_TERRAGOV)
-		if(current_timer)
-			balloon_alert(user, "You begin to stop the activation process!")
-			if(!do_after(user, deactivate_time, TRUE, src))
-				return
-			if(activated)
-				balloon_alert(user, "This sensor tower is already fully activated, you cannot deactivate it!")
-				return
-			balloon_alert(user, "You stop the activation process!")
-			deactivate()
-		else if(activated)
-			balloon_alert(user, "This sensor tower is already fully activated, you cannot deactivate it!")
-		else
-			balloon_alert(user, "This sensor tower is not activated yet, don't let it be activated!")
+	if(user.faction == faction)
+		defender_interaction(user)
+	else
+		attacker_interaction(user)
+
+///Handles defender interactions with the tower
+/obj/structure/sensor_tower_patrol/proc/defender_interaction(mob/living/user)
+	if(!activated && !current_timer)
+		balloon_alert(user, "This sensor tower is not activated yet, don't let it be activated!")
 		return
 	if(activated)
-		balloon_alert(user, "This sensor tower is already fully activated!")
+		balloon_alert(user, "This sensor tower is already fully activated, you cannot deactivate it!")
 		return
-	if(current_timer)
-		balloon_alert(user, "This sensor tower is currently activating!")
+
+	balloon_alert(user, "You begin to stop the activation process!")
+	if(!do_after(user, deactivate_time, TRUE, src))
 		return
-	if(already_activated)
-		balloon_alert(user, "There's already a sensor tower being activated!")
+	if(activated)
```

Comment:
Is being able to start deactivating when there isn't enough time left to actually deactivate it desirable?